pandas/core/arrays/string_.py

```diff
@@ -274,7 +274,16 @@ def astype(self, dtype, copy=True):
         return super().astype(dtype, copy)
 
     def _reduce(self, name, skipna=True, **kwargs):
-        raise TypeError(f"Cannot perform reduction '{name}' with string dtype")
+        if name in ["min", "max", "sum"]:
+            na_mask = isna(self)
```

Comment:
the masking should be done *inside* the methods themselves, _reduce just dispatches

Reply:
Should we implement these methods for StringArray in that case? The NA handling for PandasArray seems to be broken for string inputs, so it might have to get handled within each method

Reply:
Yes, I would say don't care about PandasArray too much, and just implement the methods here on StringArray.

Reply:
I think the reason why the NA-handling wasn't working was due to an apparently long-standing bug in nanops.nanminmax which I think we can fix here: #18588. Basically we are filling NA with infinite values when taking the min or max, but this doesn't make sense for object dtypes and an error gets raised even if skipna is True.

If we fix that by explicitly masking the missing values instead, I believe we can just use this function directly in StringArray methods.